src/Types.jl

```diff
@@ -245,7 +245,23 @@ Base.:(==)(t1::PackageEntry, t2::PackageEntry) = t1.name == t2.name &&
     t1.uuid == t2.uuid
     # omits `other`
 Base.hash(x::PackageEntry, h::UInt) = foldr(hash, [x.name, x.version, x.path, x.pinned, x.repo, x.tree_hash, x.deps, x.uuid], init=h)  # omits `other`
-const Manifest = Dict{UUID,PackageEntry}
+
+Base.@kwdef mutable struct Manifest
+    deps::Dict{UUID,PackageEntry} = Dict{UUID,PackageEntry}()
```

Comment:
Also add a dict of top level keys. Cannot be strict/limited

Reply:
Is this still a TODO?

Reply:
Sorry, missed this. After the Pkg call, I couldn't recall why it was necessary for this to be a dict rather than specific fields.
Also, I don't have much free time/bandwidth during the working week at the moment, and I'm aware this is blocking the 1.7 cutoff

I'd happily go with whatever you / @StefanKarpinski think is the right thing to do here

Reply:
I implemented this